frontend/javascripts/admin/user/permissions_and_teams_modal_view.tsx

```diff
@@ -176,60 +146,56 @@ class PermissionsAndTeamsModalView extends React.PureComponent<TeamRoleModalProp
     });
     Promise.all(newUserPromises).then(
       (newUsers) => {
-        this.props.onChange(newUsers);
+        onChange(newUsers);
       },
       () => {
         // do nothing and keep modal open
       },
     );
-  };
-  handlePermissionChanged = (evt: RadioChangeEvent) => {
+  }
+
+  function handlePermissionChanged(evt: RadioChangeEvent) {
     const selectedPermission: PERMISSIONS = evt.target.value;
-    this.setState({
-      selectedPermission,
-    });
-  };
+    setSelectedPermission(selectedPermission);
+  }
 
-  handleSelectTeamRole(teamName: string, isTeamManager: boolean) {
-    const team = this.state.teams.find((t) => t.name === teamName);
+  function handleSelectTeamRole(teamName: string, isTeamManager: boolean) {
+    const team = teams.find((t) => t.name === teamName);
 
     if (team) {
       const selectedTeam = {
         id: team.id,
         name: teamName,
         isTeamManager,
       };
-      this.setState((prevState) => ({
-        selectedTeams: update(prevState.selectedTeams, {
```

Comment:
Removed the `update` method from ìmmutability-helpers` lib. Vanilla JS can do this too.